pkg/controller/nodeipam/ipam/multinetwork_cloud_cidr_allocator.go

```diff
@@ -75,7 +81,58 @@ func (ca *cloudCIDRAllocator) PerformMultiNetworkCIDRAllocation(node *v1.Node, i
 			}
 		}
 	}
-	return defaultNwCIDRs, northInterfaces, additionalNodeNetworks, nil
+	if err = updateAnnotations(node, northInterfaces, additionalNodeNetworks); err != nil {
+		return nil, err
+	}
+	return defaultNwCIDRs, nil
+}
+
+func updateAnnotations(node *v1.Node, northInterfaces networkv1.NorthInterfacesAnnotation, additionalNodeNetworks networkv1.MultiNetworkAnnotation) error {
+	northInterfaceAnn, err := networkv1.MarshalNorthInterfacesAnnotation(northInterfaces)
+	if err != nil {
+		klog.ErrorS(err, "Failed to marshal the north interfaces annotation for multi-networking", "nodeName", node.Name)
+		return err
+	}
+	additionalNodeNwAnn, err := networkv1.MarshalAnnotation(additionalNodeNetworks)
+	if err != nil {
+		klog.ErrorS(err, "Failed to marshal the additional node networks annotation for multi-networking", "nodeName", node.Name)
+		return err
+	}
+	if node.Annotations == nil {
+		node.Annotations = make(map[string]string)
+	}
+	node.Annotations[networkv1.NorthInterfacesAnnotationKey] = northInterfaceAnn
+	node.Annotations[networkv1.MultiNetworkAnnotationKey] = additionalNodeNwAnn
+	node.Status.Capacity, err = allocateIPCapacity(node, additionalNodeNetworks)
+	return err
+}
+
+// allocateIPCapacity updates the extended IP resource capacity for every non-default network on the node.
+func allocateIPCapacity(node *v1.Node, nodeNetworks networkv1.MultiNetworkAnnotation) (v1.ResourceList, error) {
+	resourceList := node.Status.Capacity
+	if resourceList == nil {
+		resourceList = make(v1.ResourceList)
+	}
+	// Rebuild the IP capacity for all the networks on the node by deleting the existing IP capacities first.
+	for name := range resourceList {
+		if strings.HasPrefix(name.String(), networkv1.NetworkResourceKeyPrefix) {
+			delete(resourceList, name)
+		}
+	}
+	for _, nw := range nodeNetworks {
+		_, ipNet, err := net.ParseCIDR(nw.Cidrs[0])
```

Comment:
do we have guarantees that len(new.Cidrs) is always >0 ? 
can this panic?

Reply:
it is guaranteed by this [line](https://github.com/kubernetes/cloud-provider-gcp/pull/512/files#diff-3483eb4a71dd4067c8ca8685d04b9e1fc3e37178710477ae53bebec1e51ac3afR78), but I will add if condition as well.